pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -244,13 +245,19 @@ private void startReceivingMessages(List<ConsumerImpl<T>> newConsumers) {
     }
 
     private void receiveMessageFromConsumer(ConsumerImpl<T> consumer) {
-        consumer.receiveAsync().thenAcceptAsync(message -> {
+        CompletableFuture<List<Message<T>>> messagesFuture;
+        if (consumer.numMessagesInQueue() >= 10) {
```

Comment:
Since it is not configurable, should we test out a good threshold, 10 is the best?

Reply:
Can we add a new `batchReceive`  implementation to directly get as many messages in `incomingMessages`? This way you don't need to judge the size of the `numMessagesInQueue`.

Similar to just running this logic.
``` java
                MessagesImpl<T> messages = getNewMessagesImpl();
                Message<T> msgPeeked = incomingMessages.peek();
                while (msgPeeked != null && messages.canAdd(msgPeeked)) {
                    Message<T> msg = incomingMessages.poll();
                    if (msg != null) {
                        messageProcessed(msg);
                        if (!isValidConsumerEpoch(msg)) {
                            msgPeeked = incomingMessages.peek();
                            continue;
                        }
                        Message<T> interceptMsg = beforeConsume(msg);
                        messages.add(interceptMsg);
                    }
                    msgPeeked = incomingMessages.peek();
                }
```